eessi/testsuite/tests/apps/osu.py

```diff
@@ -51,39 +51,56 @@ class EESSI_OSU_Micro_Benchmarks_pt2pt(osu_benchmark):
     # unset num_tasks_per_node from the hpctestlib.
     num_tasks_per_node = None
 
+    @run_after('init')
+    def filter_scales_2gpus(self):
+        """Filter out scales with < 2 GPUs if running on GPUs"""
+        if (
+            self.device_type == DEVICE_TYPES[GPU]
+            and SCALES[self.scale]['num_nodes'] == 1
+            and SCALES[self.scale].get('num_gpus_per_node', 2) < 2
+        ):
+            self.valid_systems = [INVALID_SYSTEM]
+            log(f'valid_systems set to {self.valid_systems} for scale {self.scale} and device_type {self.device_type}')
+
+    @run_after('init')
+    def filter_benchmark_pt2pt(self):
+        """ Filter out all non-mpi.pt2pt benchmarks """
+        if not self.benchmark_info[0].startswith('mpi.pt2pt'):
+            self.valid_systems = [INVALID_SYSTEM]
+
     @run_after('init')
     def run_after_init(self):
         """hooks to run after init phase"""
-        # Note: device_buffers variable is inherited from the hpctestlib class and adds options to the launcher
-        # commands (before setup) if not equal to 'cpu'. We set it to 'cpu' initially and change it later in this hook depending on the test.
-        self.device_buffers = 'cpu'
+
         # Filter on which scales are supported by the partitions defined in the ReFrame configuration
         hooks.filter_supported_scales(self)
 
         hooks.filter_valid_systems_by_device_type(self, required_device_type=self.device_type)
-        is_cuda_module = utils.is_cuda_required_module(self.module_name)
-        # This part of the hook is meant to be for the OSU cpu tests. This is required since the non CUDA module should
-        # be able to run in the GPU partition as well. This is specific for this test and not covered by the function
-        # above.
-        if is_cuda_module and self.device_type == DEVICE_TYPES[GPU]:
-            # Sets to cuda as device buffer only if the module is compiled with CUDA.
+
+        hooks.set_modules(self)
+
+        # Set scales as tags
+        hooks.set_tag_scale(self)
+
+    @run_after('init')
+    def set_device_buffers(self):
+        """
+        device_buffers is inherited from the hpctestlib class and adds options to the launcher
+        commands in a @run_before('setup') hook if not equal to 'cpu'.
+        Therefore, we must set device_buffers *before* the @run_before('setup') hooks.
+        """
+        if self.device_type == DEVICE_TYPES[GPU]:
```

Comment:
note: checking for `device_type` is enough here, as the `is_cude_module` check is already done in `hooks.filter_valid_systems_by_device_type`